### DISCUSSION TITLE
Does not correctly export transactions if '\n' is in the Description or Original Description column

### DISCUSSION THREAD

**Discussion by bmusat** [hammem/monarchmoney#59] [General]
I've noticed that when exporting transactions from Mint, if there are any line feeds in the Description or Original Description fields in the .csv that it will not export properly.  The fields are double quoted but the '\n' in those fields is not escaped properly and hence causes problems in the exported file.

**Comment by grablair** [comment]
Hi @bmusat! This is a repository for a community-made python client to the Monarch API. We have no control over the API itself, the Monarch website, or any browser extensions Monarch provides to customers.

This sounds like a report of a potential bug in the Monarch Mint Exporter extension. To report such a bug to Monarch, you should use the "Help & Support" -> "Contact Support" option in the Monarch Website sidebar.

---

That being said, I can try and help best I can. I will be converting this into a discussion as it is not an issue with the code in this repository.

I just tried creating a transaction in Mint with the description "Test\nItem", which exported successfully using both the Mint website's "Export transactions" button, as well as the Monarch Mint Data Exporter extension. The result was:

```
"Date","Description","Original Description","Amount","Transaction Type","Category","Account Name","Labels","Notes"
"1/04/2024","Test\nItem","Test\nItem","123.00","debit","Shopping","Cash","",""
```

Which imported successfully on Monarch, so I am unable to reproduce the issue.

Are you receiving an error when importing the transactions into Monarch? If so, can you attach a trimmed CSV containing some of your problem transaction(s)?

**Reply by bmusat** [reply]
Thanks for responding.  I'm not exactly sure where the problem lies, but it could have been with the csv file I exported from the Mint Exporter extension.  I never imported it into Monarch, so it might have worked in Monarch.  I honestly was just using it to get all my historical data from Mint to use for future import into whatever financial software I end up using for 2024.  Regardless, it's not a major problem for me, but I was just reaching out to let you know of a potential problem.  

I didn't know that "Descriptions" from financial institutions could have new line characters in them, and maybe my csv reader in VSCode wasn't reading it properly.  I was worried that '\n' in that field would cause problems on linux/unix systems with '\n' for end of line, but might work on windows since they use '\r\n'.  But if every field is enclosed in "double quotes" then those escape characters in descriptions shouldn't cause problems.  

I honestly didn't investigate it further because of CSV files being so finicky and the difficultly of ensuring the encoding and decoding of them.  I'm pretty busy and this is a minor issue, but as I look closer and my exported Mint 2023 data maybe I'll stumble upon the problem again and be able to get some more information for you.  I'll post back here if I encounter it again.
